sql/txn.go

```diff
@@ -30,6 +30,9 @@ func (p *planner) BeginTransaction(n *parser.BeginTransaction) (planNode, error)
 // CommitTransaction commits a transaction.
 func (p *planner) CommitTransaction(n *parser.CommitTransaction) (planNode, error) {
 	err := p.txn.Commit()
+	if err != nil {
+		_ = p.txn.Rollback()
```

Comment:
Yes, that's correct, assuming that's what we wound up implementing.

On Tue, Aug 25, 2015 at 1:55 PM, Peter Mattis <notifications@github.com>
wrote:

> In sql/txn.go
> <https://github.com/cockroachdb/cockroach/pull/2260#discussion_r37896826>:
>
> > @@ -30,6 +30,9 @@ func (p *planner) BeginTransaction(n *parser.BeginTransaction) (planNode, error)
> >  // CommitTransaction commits a transaction.
> >  func (p *planner) CommitTransaction(n *parser.CommitTransaction) (planNode, error) {
> >  	err := p.txn.Commit()
> > +	if err != nil {
> > +		_ = p.txn.Rollback()
>
> I thought we had agreed that if a commit failed a rollback wouldn't be
> necessary? @tschottdorf <https://github.com/tschottdorf>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2260/files#r37896826>.
>